### ISSUE TITLE
Connectivity Service protection types

### ISSUE THREAD

**Issue by arthurMll** [OpenNetworkingFoundation/TAPI#334]
Hi,

I have some doupts regarding the representation of some standard protection mechanims with TAPI.
In particular I would like to know if there is any documentation so far on how to use the models to configure different common protection schemas such OLP and E-SNCP.

In the case a protection mechanism such OLP use a single signal source  the correct protection type mechanism in tapi would be ONE_FOR_ONE_PROTECTION?

I think its needed to add the descriptions to the model definitions within typedef protection-type inside tapi-topology model, to clarify its applicability.

**Comment by jgroom33** [comment]
Agreed.  The following needs definitions:
          - NO_PROTECTON
          - ONE_PLUS_ONE_PROTECTION
          - ONE_PLUS_ONE_PROTECTION_WITH_DYNAMIC_RESTORATION
          - PERMANENT_ONE_PLUS_ONE_PROTECTION
          - ONE_FOR_ONE_PROTECTION
          - DYNAMIC_RESTORATION
          - PRE_COMPUTED_RESTORATION

**Comment by arthurMll** [comment]
Additionally, I think there is missing the specification of the distinction between the nominal and protection connections supporting a protected connectivity service of the following types:
- ONE_PLUS_ONE_PROTECTION
- ONE_PLUS_ONE_PROTECTION_WITH_DYNAMIC_RESTORATION
- PERMANENT_ONE_PLUS_ONE_PROTECTION
- ONE_FOR_ONE_PROTECTION

, additionally for services with protection type PRE_COMPUTED_RESTORATION, there is missing an attribute which distinguish the nominal path and the secondary (restoration) path, within the route attribute of the connection object.

Reviewing the document I found:
```
leaf connection-port-role {
                type tapi-common:port-role;
                config false;
                description "Each EP of the FC has a role (e.g., working, protection, protected, symmetric, hub, spoke, leaf, root)  in the context of the FC with respect to the FC function. ";
            }
```
which states for the differentiation of connection-end-point roles, however I haven't found the types WORKING and PROTECTION in  tapi-common@2018-08-31.yang model, currently these are the only roles defined:
```
typedef port-role {
            type enumeration {
                enum SYMMETRIC {
                    description "none";
                }
                enum ROOT {
                    description "none";
                }
                enum LEAF {
                    description "none";
                }
                enum TRUNK {
                    description "none";
                }
                enum UNKNOWN {
                    description "none";
                }
            }
            description "The role of an end in the context of the function of the forwarding entity that it bounds";
}
```

**Comment by RodLU** [comment]
You can check if the following link helps.
https://wiki.opennetworking.org/download/attachments/259719184/otcc2018.Rod.003-TAPI_Resilience_use_case.v02.docx?api=v2

**Comment by amazzini** [comment]
Solved by 2.1.3 and 2.3.1 versions.
